packages/amplify-codegen/src/commands/models.js

```diff
@@ -72,6 +73,9 @@ async function generateModels(context) {
   const generateIndexRules = readFeatureFlag('codegen.generateIndexRules');
   const emitAuthProvider = readFeatureFlag('codegen.emitAuthProvider');
 
+  const withNullSafety =
+    projectConfig.frontend === 'flutter' ? validateDartSDK(context, projectRoot) && readFeatureFlag('codegen.withNullSafety') : false;
```

Comment:
can we modify the FF to `enableDartNullSafety` as per https://github.com/aws-amplify/amplify-cli/pull/7516 just to be more clear that it is Dart specific.